hr_holidays_leave_auto_approve/models/hr_holidays.py

```diff
@@ -0,0 +1,20 @@
+# -*- coding: utf-8 -*-
+# © 2016 ONESTEiN BV (<http://www.onestein.eu>)
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+from openerp import models, api
+
+
+class HrHolidays(models.Model):
+    _inherit = "hr.holidays"
+
+    @api.model
+    @api.returns('self', lambda value: value.id)
+    def create(self, values):
+        result = super(HrHolidays, self).create(values)
+        if values.get('holiday_status_id'):
+            type_obj = self.env['hr.holidays.status']
+            type = type_obj.browse(values.get('holiday_status_id'))
+            if type.auto_approve:
+                result.sudo().holidays_validate()
```

Comment:
@sylvain-garancher ,
About using the workflow process you are right. I fixed this in the latest commit.

About using `sudo`:
Unfortunately, relying on the workflow mechanism makes impossible to validate a leave request if the user is not at least an HR officer (group: `base.group_hr_user`). That's because the workflow transition from the state `confirm` when a signal `validate` is received is defined as:
```

<record model="workflow.transition" id="holiday_confirm2validate"> <!-- 2. submitted->accepted (validate signal) if not double_validation-->
        <field name="act_from" ref="act_confirm" />
        <field name="act_to" ref="act_validate" />
        <field name="signal">validate</field>
        <field name="condition">not double_validation</field>
        <field name="group_id" ref="base.group_hr_user"/>
```
A user not belonging to the group `base.group_hr_user` is not allowed to "see" this transition, therefore its signal doesn't trigger anything. A normal user is not allowed to approve a leave request and that can't (and shouldn't) be bypassed.

Apart from that, using `sudo` (which conceptually means designating the system rather than the user as validator) seems more correct to me. Indeed it's the system and not the requesting user to approve the leave request, following a specific configuration setted on the leave type by an HR officer. In my opinion, the requesting user shouldn't absolutely be involved into the approval process (e.g.: the message tracking the state change shouldn't be authored by the user).